lib/datadog/tracing/sampling/span/rule.rb

```diff
@@ -22,7 +22,7 @@ class Rule
           def initialize(
             matcher,
             sample_rate: Span::Ext::DEFAULT_SAMPLE_RATE,
-            rate_limit: Span::Ext::DEFAULT_MAX_PER_SECOND
+            rate_limit: nil
```

Comment:
Interesting. Why demote this from constructor args to constructor body?

Reply:
I need to know when `rate_limit` was not set (thus `nil`).

Before, it would fall back to the default unlimited value `-1`, but when I can't tell if the user has explicitly set `-1` or not set anything at all.

We only want to remove the `span_op.set_metric(Span::Ext::TAG_MAX_PER_SECOND, ...` tag if the user did not provide a `rate_limit`.